src/query/storage/promremote/storage.go

```diff
@@ -195,36 +189,31 @@ type promStorage struct {
 	scope           tally.Scope
 	droppedWrites   tally.Counter
 	enqueued        tally.Counter
+	enqueueErr      tally.Counter
+	dupWrites       tally.Counter
+	ingestorWrites  tally.Counter
 	batchWrite      tally.Counter
 	batchWriteErr   tally.Counter
 	tickWrite       tally.Counter
 	logger          *zap.Logger
 	queryQueue      chan *storage.WriteQuery
 	workerPool      xsync.WorkerPool
-	pendingQuery    map[tenant]*WriteQueue
+	pendingQuery    map[tenantKey]*WriteQueue
 	noTenantFound   tally.Counter
 	cancel          context.CancelFunc
 	writeLoopDone   chan struct{}
+	wrongTenant     tally.Counter
 }
 
-type tenant struct {
-	name string
-	attr storagemetadata.Attributes
-}
+type tenantKey string
```

Comment:
wonder if this affects the discrepancy for ingested samples
![Uploading Screenshot 2024-04-30 at 8.45.00 PM.png…]()

Reply:
actually this was there before the latest change